pkg/ingester/client/compat.go

```diff
@@ -256,3 +256,24 @@ func LabelsToKeyString(l labels.Labels) string {
 	b := make([]byte, 0, 1024)
 	return string(l.Bytes(b))
 }
+
+// CopyTimeSeriesChunk makes a deep copy of a TimeSeriesChunk object.
+func CopyTimeSeriesChunk(ts TimeSeriesChunk) TimeSeriesChunk {
+	lbs := make([]mimirpb.LabelAdapter, len(ts.Labels))
+	copy(lbs, ts.Labels)
+
+	chunks := make([]Chunk, len(ts.Chunks))
+	for j := 0; j < len(ts.Chunks); j++ {
```

Comment:
It wouldn't be easier/equivalent (plus you won't have to state each chunk field explicitly) here to just copy `ts.Chunks`? Let me know if I didn't think of something :)

```suggestion
	copy(chunks, ts.Chunks)
	for c, j := range ts.Chunks {
		chunks[j].Data = make([]byte, len(c.Data))
		copy(chunks[j].Data, c.Data)
	}
```

Reply:
Problem with that is that we wouldn't be creating a deep copy of the original `TimeSeriesChunks` object, since `Data` backed buffer would be shared between source and destination chunks, which can cause some unexpected behaviors when reusing those from pool.

Reply:
Oh, sorry. I misread it completely. 🤦‍♂️  

And yes, I agree, that's equivalent and more concise. 👍

Reply:
Yay :D